tests/core/selection/selection.js

```diff
@@ -785,5 +786,22 @@ bender.test( {
 		bender.tools.setHtmlWithSelection( editor, '<p>T^es^t</p>' );
 
 		assert.isFalse( editor.getSelection().isCollapsed() );
+	},
+
+	'test delete/backspace keys are not removing readonly selection': function() {
+		var editor = this.editor, bot = this.editorBot;
+
+		editor.setReadOnly( true );
+
+		bot.setData( '<p>[[placeholder]]</p>', function() {
```

Comment:
Please restore editor readOnly state back to `false` as any subsequent TCs would be fired on readonly instance (which is unexpected).